### DISCUSSION TITLE
how to support mysql/sqlserver/postgres

### DISCUSSION THREAD

**Discussion by formcms** [fluentcms/FluentCMS#1850] [Ideas]
I think what you are doing is great.

As I understanding, all content is save same collection distinguish by a discriminator with content_type/content_slug.

That is great idea since no sql have no fixed schema. I almost think all my efforts are wasted as I am trying to do a similar project, but I am using rdbms, so I have to create a table for each entity.   my repo's link https://github.com/fluent-cms/fluent-cms

while nosql is nice, but it might be a issue to ensure data integrity.  e.g. in RDBMS there are 1 to 1, 1 to man, many to many relationships.   e.g. post, tag, post_tag. I kind done that for your reference if you need. 

I created a nuget package for it if you want to have a look

**Comment by pournasserian** [comment]
Thanks @fluent-cms 

As you see, we have a repository layer and we have considered implementing EF layer to support SQLServer, SQLite, MySQL.

But, our main focus is to provide an MVP for the first phase. Then, that would be in our roadmap.

That would be great if you raise PR for any features to improve functionalities.
